### PR TITLE
completes Q4 GET articles

### PR DESCRIPTION
**Pull request by robbochobbo**
Creates GET /api/articles/:article_id endpoint and includes error handling

### CODE REVIEW

**Comment by robbochobbo** [comment]
I've just realised that I got confused with my SQL query! I thought the only reference to the article_id was in the comments table so I did a JOIN. But I could just do:

SELECT 
*
FROM articles
WHERE articles.article_id = $1;

So I'm going to fix that now and make a new pull request called Q4-GET-articles-by-id-update

File: models/models.js [hunk 1]
```diff
@@ -25,10 +25,42 @@ const fetchAllTopics = (sort_by='slug', order='ASC', slug) => {
...
+    articles.topic,
+    articles.created_at,
+    articles.votes,
+    articles.article_img_url
+    FROM articles
+    JOIN comments
+    ON articles.article_id = comments.article_id
+    WHERE comments.article_id = $1;`, [article_id])
+    .then((body) => {
+      if(body.rows.length === 0){
```
Comment (nc-saima-khanom) [inline models/models.js:56]: Nice!

File: endpoints.json [hunk 2]
```diff
@@ -25,5 +25,32 @@
         }
       ]
     }
+  },
+  "GET /api/articles/:article_id": {
+    "description": "serves an array of all articles of a given article_id",
+    "exampleResponse": {
+        "articles": [
```
Comment (nc-saima-khanom) [inline endpoints.json:32]: should we be sending back an array of articles or one article object?

File: endpoints.json [hunk 3]
```diff
@@ -25,5 +25,32 @@
         }
       ]
     }
+  },
+  "GET /api/articles/:article_id": {
+    "description": "serves an array of all articles of a given article_id",
```
Comment (nc-saima-khanom) [inline endpoints.json:30]: does each article id have multiple articles?

File: models/models.js [hunk 4]
```diff
@@ -25,10 +25,42 @@ const fetchAllTopics = (sort_by='slug', order='ASC', slug) => {
...
+        return Promise.reject({status: 404, msg:'Article does not exist'})
+      }
+      return body.rows;
```
Comment (nc-saima-khanom) [inline models/models.js:59]: as we are only trying to find and send back a particular article, we dont need to send the whole array back here

File: endpoints.json (hunk 2 above)
Comment (robbochobbo) [reply endpoints.json:32]: Thanks saima - please check the updated pull request :)

**Pull request merged**